### PR TITLE
Setup PM2

### PR DESCRIPTION
**Pull request by DesignrKnight**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

### Description
Changes in the PR
- Setup PM2 for development and production, including Docker
- Setup PM2 keymetrics for remote monitoring. Refer: https://pm2.io/
- Setup logging to file in production and map to volume in Docker
- Have a separate file for local and server docker-compose. Reason: Server pulls image and runs, local builds image and runs
- Added a .env in server to help monitor it remotely. Not needed on local we do not want remote logging for local
- Setup cross env to help in using env variables consistently across npm scripts in both Linux and Windows system. Refer: https://www.npmjs.com/package/cross-env

❤️Thank you!

### Post merge checklist
- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

**Comment by DesignrKnight** [comment]
Merge this after #84 to be able to confirm if push trigger works

**Comment by DesignrKnight** [comment]
Changes
1. Docker restart
2. PM2 instances to 1 on dev

**Comment by DesignrKnight** [comment]
PR ready for review. @HarishTeens @roshankshaw

File: sources/package.json [hunk 1]
```diff
@@ -6,11 +6,16 @@
...
+		"prettier:check": "npx prettier --check .",
+		"eslint": "npx eslint",
+		"eslint:fix": "npx eslint --fix",
+		"start": "npx pm2 start ecosystem.config.js --only dev --env development",
+		"stop": "npx pm2 stop ecosystem.config.js --only dev --env development",
+		"delete": "npx pm2 delete ecosystem.config.js --only dev",
+		"restart": "npx pm2 restart ecosystem.config.js --only dev --env development",
+		"start:prod": "pm2-runtime start ecosystem.config.js --only prod --env production",
+		"test": "cross-env NODE_ENV=test jest --runInBand --coverage",
+		"test:watch": "cross-env NODE_ENV=test --runInBand --watch"
```
Comment (HarishTeens) [inline sources/package.json:18]: @DesignrKnight Can we have a status script here? Even pm2 status wont work

Comment (DesignrKnight) [reply sources/package.json]: Done in https://github.com/dscnitrourkela/project-dates/pull/82/commits/cb976323395b63c7b0ee4e57241c0aad357bc68a

File: sources/package.json [hunk 2]
```diff
@@ -13,6 +13,8 @@
 		"stop": "npx pm2 stop ecosystem.config.js --only dev --env development",
 		"delete": "npx pm2 delete ecosystem.config.js --only dev",
 		"restart": "npx pm2 restart ecosystem.config.js --only dev --env development",
+		"logs": "npx pm2 logs",
+		"status": "npx pm2 ps",
```
Comment (HarishTeens) [inline sources/package.json:17]: Thank you ⭐

**Pull request merged**